src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -2,18 +2,47 @@
 
 import core.basesyntax.Storage;
 
+@SuppressWarnings("unchecked")
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int Capacity = 10;
+    private K[] keys;
+    private V[] values;
+    private int size;
+
+    public StorageImpl() {
+        keys = (K[]) new Object[Capacity];
+        values = (V[]) new Object[Capacity];
+        size = 0;
+    }
+
     @Override
     public void put(K key, V value) {
+        for (int i = 0; i < size(); i++) {
+            if ((keys[i] == key) || (key != null && key.equals(keys[i]))) {
```

Comment:
```suggestion
            if (keys[i] == key || key != null && key.equals(keys[i])) {
```
Redundant brackets (read about operator priority)